sklearn/linear_model/ridge.py

```diff
@@ -545,6 +545,24 @@ def fit(self, X, y, sample_weight=None):
                          accept_sparse=_accept_sparse,
                          dtype=_dtype,
                          multi_output=True, y_numeric=True)
+        if sparse.issparse(X) and self.fit_intercept:
+            if self.solver not in ['auto', 'sparse_cg', 'sag']:
+                raise ValueError(
+                    "solver='{}' does not support fitting the intercept "
+                    "on sparse data. Please set the solver to 'auto' or "
+                    "'sparse_cg', or set `fit_intercept=False`"
+                    .format(self.solver))
+            if self.solver == 'sag':
+                warnings.warn(
+                    '"sag" solver requires many iterations to fit '
+                    'an intercept with sparse inputs. Either set the '
+                    'solver to "auto" or "sparse_cg", or set a low '
+                    '"tol" and a high "max_iter".')
```

Comment:
what bothers me is that you will get the warning whatever you use for tol or max_iter. Maybe just warn if parameter used are the defaults?